source/slang/preprocessor.cpp

```diff
@@ -2268,8 +2268,8 @@ static void DefineMacro(
 
     auto sourceManager = preprocessor->translationUnit->compileRequest->getSourceManager();
 
-    SourceFile* keyFile = sourceManager->createSourceFile(pathInfo, key);
-    SourceFile* valueFile = sourceManager->createSourceFile(pathInfo, value);
+    RefPtr<SourceFile> keyFile = sourceManager->createSourceFileWithString(pathInfo, key);
```

Comment:
When did `SourceFile` switch to being reference-counted? It seems like it shouldn’t need it, since all source files should be owned by the `SourceManager` and should never out-live it.